src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -223,26 +219,42 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 				continue;
 			}
 			const isDirectory = kind === TerminalCompletionItemKind.Folder;
-			const pathToResource = stat.resource.fsPath.replace(cwd.fsPath, '');
-			let label = pathToResource;
+			const pathToResource = resourceRequestConfig.pathSeparator + basename(stat.resource.fsPath);
+
+			let label;
+			if (!lastWord.startsWith('.' + resourceRequestConfig.pathSeparator) && !lastWord.startsWith('..' + resourceRequestConfig.pathSeparator)) {
+				// add a dot to the beginning of the label if it doesn't already have one
+				label = '.' + pathToResource;
+			} else {
+				if (lastWord.endsWith(resourceRequestConfig.pathSeparator)) {
+					// prevent a double path separator
+					label = lastWord + pathToResource.substring(1);
+				} else {
+					label = lastWord + pathToResource;
+				}
```

Comment:
`resourceRequestConfig.pathSeparator` is added above at line 222, this code would be easier to follow without the undoing of that. So something like this instead:

```ts
const fileName = basename(stat.resource.fsPath);
...
if (lastWord.endsWith(resourceRequestConfig.pathSeparator)) {
  label = `${lastWord}${fileName}`;
} else {
  label = `${lastWord}${resourceRequestConfig.pathSeparator}${fileName}`;
}
```